### PR TITLE
Update to latest version of matrix-js-sdk and convert to TypeScript

### PR DESCRIPTION
**Pull request by Johennes**
This updates the SDK to the latest version of matrix-js-sdk, adds missing methods in the crypto store and converts everything from JS to TS.

### CODE REVIEW

File: src/AsyncCryptoStore.ts
Comment (t3chguy) [inline src/AsyncCryptoStore.ts:1]: Missing copyright

File: src/AsyncStore.ts
Comment (t3chguy) [inline src/AsyncStore.ts:1]: Missing copyright

**Review by t3chguy — APPROVED** [review]
Looks sane otherwise

File: src/AsyncCryptoStore.ts
Comment (Johennes) [reply src/AsyncCryptoStore.ts:1]: Argh, thanks. I always forget about these. 🤦‍♂️

**Review by t3chguy — APPROVED** [review]

**Comment by t3chguy** [comment]
Turns out my review doesn't mean squat in this repo, no clue who has permissions here.

**Comment by Johennes** [comment]
@dbkr possibly?

**Review by dbkr — APPROVED** [review]

**Pull request merged**